chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java

```diff
@@ -96,24 +103,29 @@
                     + "the following: 1) /invite 2) /create or /authorize 3) /confirm ")
     @RequestMapping(value = "/create", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
     produces = "application/json; charset=utf-8")
-    public User createUserDeprecated(@RequestBody final CreateUserFromInvitationRequest userInfo)
-            throws InvalidArgumentsException, UserCreationException, UserRetrievalException, EntityRetrievalException,
-            MessagingException, JsonProcessingException, EntityCreationException {
+    public @ResponseBody User createUserDeprecated(@RequestBody final CreateUserFromInvitationRequest userInfo)
+            throws ValidationException, EntityRetrievalException, InvalidArgumentsException, UserRetrievalException,
+            UserCreationException, MessagingException, JsonProcessingException, EntityCreationException {
 
         return create(userInfo);
     }
 
     private User create(final CreateUserFromInvitationRequest userInfo)
-            throws InvalidArgumentsException, UserCreationException, UserRetrievalException, EntityRetrievalException,
-            MessagingException, JsonProcessingException, EntityCreationException {
+            throws ValidationException, EntityRetrievalException, InvalidArgumentsException, UserRetrievalException,
+            UserCreationException, MessagingException, JsonProcessingException, EntityCreationException {
 
         if (userInfo.getUser() == null || userInfo.getUser().getSubjectName() == null) {
-            throw new InvalidArgumentsException("Username ('subject name') is required.");
+            throw new ValidationException("Username ('subject name') is required.");
+        }
+
+        Set<String> errors = validateCreateUserFromInvitationRequest(userInfo);
+        if (errors.size() > 0) {
+            throw new ValidationException(errors, null);
         }
 
         InvitationDTO invitation = invitationManager.getByInvitationHash(userInfo.getHash());
         if (invitation == null || invitation.isOlderThan(VALID_INVITATION_LENGTH)) {
-            throw new InvalidArgumentsException(
+            throw new ValidationException(
                     "Provided user key is not valid in the database. The user key is valid for up to 3 days from "
```

Comment:
Since you touched this line can you do the same with the errors properties file?

Reply:
Fixed